### DISCUSSION TITLE
'kaniko-project/executor' container image is different between amd64 and arm64 in build process

### DISCUSSION THREAD

**Discussion by thuanghai** [strimzi/strimzi-kafka-operator#5488] [Q&A]
Thankyou for your help, I have build the container images for amd64 successfullly some days ago. Now I start build images for arm64. And I have a question, whether the container image -- "gcr.io/kaniko-project/executor:v1.6.0" is the same or not when I build images in 'strimzi-kafka-operator' for both amd64 and arm64.
Because, I can't pull 'gcr.io/kaniko-project/executor:v1.6.0' directly. And I find a method to deel with this problem, but only for amd64(I am sure). So I want to know whether the executor image can be reused need when I building this project images for arm64.

**Comment by scholzj** [comment]
I'm not sure I follow that the problem is ... why can't you pull `gcr.io/kaniko-project/executor:v1.6.0`? I never used it, but IMHO the `gcr.io/kaniko-project/executor:v1.6.0` image is multiarch so if you build Strimzi on Arm64 it should pull the right one.

**Reply by thuanghai** [reply]
Building kafka image including 'kafka connect' which is builded by executor described in '<project_root_dir>/kaniko-executor/Makefile'. So kafka connector may not be deployed in worst-case. Is that so?

**Reply by scholzj** [reply]
Well, as I said, I think if you build Strimzi on Arm64, it should also take the Arm64 version of the Kaniko image. And when you use it on Arm server with Arm base image, it should build new Arm image. But as I said, I haven't tried this and not sure anyone else did. So it is up to you to try it and let us know ;-).

If it doesn't work, you do not need to use the _build_ feature - you can also build you Connect image with the connectors manually from a Dockerfile as described in the docs.

**Reply by thuanghai** [reply]
En, thank you for your answer, I check the Dockerfile and Makefile. I think you are right. I have not finded it will affect operator and kafka images up to now. I will continue test the building images for arm64.
